d8s_domains/domains.py

```diff
@@ -86,6 +86,33 @@ def domain_rank(domain_name: str) -> int:
     return get(onemillion_api_url, process_response=True)
 
 
+@get_first_arg_url_domain
+def domain_is_member(domain_to_check: str, domain_base: str) -> bool:
+    """Given two domains, check if the first domain is a member of the second domain.
+    A member means it is either the domain itself, or a subdomain of the domain.
+    """
+    if domain_to_check == domain_base:
+        return True
+
+    name = domain_second_level_name(domain_to_check)
+    tld = domain_tld(domain_to_check)
+
+    if domain_tld(domain_base) == tld and domain_second_level_name(domain_base) == name:
+        subdomain_str = domain_subdomains(domain_to_check)
+        subdomains = subdomain_str.split('.')
+        last_subdomain = subdomains[-1]
+
+        base_subdomain_str = domain_subdomains(domain_base)
+        base_subdomains = base_subdomain_str.split('.')
+        base_last_subdomain = base_subdomains[-1]
+
+        if not base_last_subdomain:
+            return domain_to_check.endswith("." + domain_base)
+        elif last_subdomain == base_last_subdomain:
+            return True
+    return False
+
+
```

Comment:
Thanks for suggesting this function! I do have a question. From what I can tell, the body of this function would be simplified to:

```python
return domain_to_check.endswith(domain_base)
```

and it would still do everything you want. Does that make sense? Are there edge-cases this change would not handle?

For example, the code snippet below shows a more simple version of the `domain_is_member` function that passes all of your test cases:

```python
def domain_is_member(domain_to_check, domain_base):
    return domain_to_check.endswith(domain_base)


assert domain_is_member('example.com', 'example.com') is True
assert domain_is_member('example.com', 'google.com') is False
assert domain_is_member('test.example.com', 'example.com') is True
assert domain_is_member('example.com', 'test.example.com') is False
assert domain_is_member('t.example.com', 'test.example.com') is False
assert domain_is_member('test1.example.com', 'test.example.com') is False
assert domain_is_member('example.com.notdomain.com', 'example.com') is False
assert domain_is_member('test.example.com.notdomain.com', 'example.com') is False
assert domain_is_member('1.test.example.com.notdomain.com', 'test.example.com') is False
assert domain_is_member('test.example.com', 'test.example.com') is True
assert domain_is_member('subdomain.test.example.com', 'test.example.com') is True
assert domain_is_member('example.com', 'testexample.com') is False
assert domain_is_member('example.com', 'test.testexample.com') is False
assert domain_is_member('test.example.com', 'test.testexample.com') is False
assert domain_is_member('example.com', 'example.com.br') is False
```

Is there an advantage to doing it the way you suggested?

Reply:
Yes, `assert domain_is_member('example.com', 'testexample.com') is False` will not pass with a simple endswith 

The purpose of this function is to check if the domain is a subdomain or the exact domain.

If it were that simple I would've done it that way
:)

Reply:
some tests were not in the correct order. I'll add more tests for clarity

Reply:
Haha, I figured I was missing something... thanks for pointing that out.